File: server/request_payload.py
class RequestUserPayload:
  def __init__(self, name):
    self.name = name

  def get_name(self):
    return self.name
  
class RequestFilePayload:
  def __init__(self, file_name):
    self.file_name = file_name

  def get_file_name(self):
    return self.file_name
  
class RequestPublicKeyPayload(RequestUserPayload):
  def __init__(self, name, public_key):
    super().__init__(name)
    self.public_key = public_key

  def get_public_key(self):
    return self.public_key
  
class RequestSentFilePayload(RequestFilePayload):
  def __init__(self, content_size, file_name, message_content):
    super().__init__(file_name)
    self.content_size = content_size
    self.message_content = message_content

  def get_content_size(self):
    return self.content_size
  
  def get_message_content(self):
    return self.message_content

File: client/request_payload.cpp
#include <cstring>
#include "request_payload.h"

RequestUserPayload::RequestUserPayload(std::string name) {
  strncpy(this->name, name.c_str(), 255);
}

std::string RequestUserPayload::get_name() {
  return std::string(this->name);
}

uint32_t RequestUserPayload::get_size() {
  return 255;
}

RequestFilePayload::RequestFilePayload(std::string file_name) {
  strncpy(this->file_name, file_name.c_str(), 255);
}

std::string RequestFilePayload::get_file_name() {
  return std::string(this->file_name);
}

uint32_t RequestFilePayload::get_size() {
  return 255;
}

RequestPublicKeyPayload::RequestPublicKeyPayload(std::string name, std::string public_key) 
  : RequestUserPayload(name) {
    strncpy(this->public_key, public_key.c_str(), 160);
}

std::string RequestPublicKeyPayload::get_public_key() {
  return std::string(this->public_key);
}

uint32_t RequestPublicKeyPayload::get_size() {
  return 250 + 160;
}

RequestSentFilePayload::RequestSentFilePayload(std::string file_name, uint32_t content_size, std::string message_content)
  : RequestFilePayload(file_name)
  , content_size(content_size)
  , message_content(message_content) {}

uint32_t RequestSentFilePayload::get_content_size() {
  return this->content_size;
}

std::string RequestSentFilePayload::get_message_content() {
  return this->message_content;
}

uint32_t RequestSentFilePayload::get_size() {
  return 255 + 4 + this->content_size;
}